MixItUp.Desktop/Services/DeveloperAPI/CommandController.cs

```diff
@@ -54,42 +46,107 @@ public Command Run(Guid commandID, [FromBody] IEnumerable<string> arguments)
             {
                 throw new HttpResponseException(HttpStatusCode.NotFound);
             }
 
             selectedCommand.IsEnabled = commandData.IsEnabled;
-            return CommandFromCommandBase(selectedCommand);
+            return CommandFromCommandBase(selectedCommand, category);
         }
 
-        private List<CommandBase> GetAllCommands()
+        private CommandBase FindCommand(Guid commandId, out string category)
         {
-            List<CommandBase> allCommands = new List<CommandBase>();
-            allCommands.AddRange(ChannelSession.Settings.ChatCommands);
-            allCommands.AddRange(ChannelSession.Settings.InteractiveCommands);
-            allCommands.AddRange(ChannelSession.Settings.EventCommands);
-            allCommands.AddRange(ChannelSession.Settings.TimerCommands);
-            allCommands.AddRange(ChannelSession.Settings.ActionGroupCommands);
-            allCommands.AddRange(ChannelSession.Settings.GameCommands);
+            category = null;
+            CommandBase command = ChannelSession.Settings.ChatCommands.SingleOrDefault(c => c.ID == commandId);
+            if (command !=null)
+            {
+                category = "Chat";
+                return command;
+            }
+
+            command = ChannelSession.Settings.InteractiveCommands.SingleOrDefault(c => c.ID == commandId);
+            if (command != null)
+            {
+                category = "Interactive";
+                return command;
+            }
+
+            command = ChannelSession.Settings.EventCommands.SingleOrDefault(c => c.ID == commandId);
+            if (command != null)
+            {
+                category = "Event";
+                return command;
+            }
+
+            command = ChannelSession.Settings.TimerCommands.SingleOrDefault(c => c.ID == commandId);
+            if (command != null)
+            {
+                category = "Timer";
+                return command;
+            }
+
+            command = ChannelSession.Settings.ActionGroupCommands.SingleOrDefault(c => c.ID == commandId);
+            if (command != null)
+            {
+                category = "ActionGroup";
+                return command;
+            }
+
+            command = ChannelSession.Settings.GameCommands.SingleOrDefault(c => c.ID == commandId);
+            if (command != null)
+            {
+                category = "Game";
+                return command;
+            }
+
+            command = ChannelSession.PreMadeChatCommands.SingleOrDefault(c => c.ID == commandId);
+            if (command != null)
+            {
+                category = "PreMade";
+                return command;
+            }
+
+            return null;
+        }
+
+        private List<Command> GetAllCommands()
+        {
+            List<Command> allCommands = new List<Command>();
+            allCommands.AddRange(CommandsFromCommandBases(ChannelSession.Settings.ChatCommands, "Chat"));
```

Comment:
If it helps, you could call ChannelSession.AllCommands() to get the list of stuff below. However, you'd still need to divide it out based on the category.